samigo/samigo-app/src/java/org/sakaiproject/tool/assessment/ui/listener/delivery/DeliveryActionListener.java

```diff
@@ -386,6 +389,11 @@ public void processAction(ActionEvent ae) throws
             	  log.debug("Get data from db otherwise");
                   // this returns a HashMap with (publishedItemId, itemGrading)
                   itemGradingHash = service.getLastItemGradingData(id, agent); //
+                  if (!itemGradingHash.isEmpty() && !studentRichText) {
+                	  itemGradingHash.entrySet().stream().map(entry ->
+                	  	((Map.Entry<Long, ArrayList>)entry).getValue()).flatMap(item ->
```

Comment:
Can't we modify getLastItemGradingData to just return the correct type, rather than having this map operation? It makes a complicated transformation more complex :)